pkg/syncer/syncers/client.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package syncers
+
+import (
+	"context"
+
+	edgev1alpha1 "github.com/kcp-dev/edge-mc/pkg/syncer/apis/edge/v1alpha1"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/dynamic"
+)
+
+type Client struct {
+	ResourceClient dynamic.NamespaceableResourceInterface
+	scope          meta.RESTScope
+}
+
+func (c *Client) IsNamespaced() bool {
+	return c.scope == meta.RESTScopeNamespace
+}
+
+func (c *Client) Create(resource edgev1alpha1.EdgeSyncConfigResource, unstObj *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+	var createdObj *unstructured.Unstructured
+	var err error
+	if c.IsNamespaced() {
+		createdObj, err = c.ResourceClient.Namespace(resource.Namespace).Create(context.Background(), unstObj, v1.CreateOptions{})
```

Comment:
I recommend filling in the `FieldManager` value for the various meta/v1/XXXOptions